assets/src/edit-story/components/fontPicker/pickerContainer.js

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useRef, useMemo, useCallback } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { rgba } from 'polished';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useFocusOut from '../../utils/useFocusOut';
+import { useFont } from '../../app/font';
+import { TextInput } from '../form';
+
+const PickerContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  width: 100%;
+  min-width: 160px;
+  max-height: 355px;
+  overflow: hidden;
+  border-radius: 4px;
+  background-color: ${({ theme }) => theme.colors.fg.v1};
+  background-clip: padding-box;
+  box-shadow: 0 6px 12px ${({ theme }) => rgba(theme.colors.bg.v0, 0.175)};
+  padding: 0;
+`;
+
+const ListContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  width: 100%;
+  max-height: 305px;
+  overflow-x: hidden;
+  overflow-y: auto;
+  overscroll-behavior: none auto;
+  padding-bottom: 8px;
+  ${({ menuFonts }) => menuFonts}
```

Comment:
Can you explain how this works? Do you constantly add/remove `@font-face` CSS rules? Did you do profiling for this? How significant is it for browser to re-insert fonts and redecode them anew each time? Does the browser always uses local cache for a reinserted font file?

Also, if you only return the "new delta" from the `getMenuFonts`, wouldn't this skip some of the fonts you need?

Reply:
I had various test cases for this one. With `createGlobalStyle` we can just add 10 visible font faces at a time. When the menuFonts changed, it will remove original ones and insert 10 new ones again. I think it's not necessary. From this approach, menuFonts is font faces and styled component will insert them in document level not for this particular class. Once it's inserted, the font face will not be removed unless we're refreshing. So each time, we can just insert new font faces.

Reply:
And yes, browser uses local cache for a reinserted font file.